airflow/providers/amazon/aws/hooks/glue_crawler.py

```diff
@@ -0,0 +1,294 @@
+    :type security_configuration = Optional[str]
+    :param tags = Tags to attach to the crawler request
+    :type tags = Optional[dict]
+    :param overwrite = Determines if crawler should be updated if the crawler configuration change
+    :type overwrite = Optional[bool]
+    """
+
+    CRAWLER_POLL_INTERVAL = 6  # polls crawler status after every CRAWLER_POLL_INTERVAL seconds
+
+    def __init__(
+        self,
+        crawler_name=None,
+        crawler_desc=None,
+        glue_db_name=None,
+        iam_role_name=None,
+        s3_targets_configuration=None,
+        jdbc_targets_configuration=None,
+        mongo_targets_configuration=None,
+        dynamo_targets_configuration=None,
+        glue_catalog_targets_configuration=None,
+        cron_schedule=None,
+        classifiers=None,
+        table_prefix=None,
+        update_behavior=None,
+        delete_behavior=None,
+        recrawl_behavior=None,
+        lineage_settings=None,
+        json_configuration=None,
+        security_configuration=None,
+        tags=None,
+        overwrite=False,
+        *args,
+        **kwargs,
+    ):
+
+        self.crawler_name = crawler_name
+        self.crawler_desc = crawler_desc
+        self.glue_db_name = glue_db_name
+        self.iam_role_name = iam_role_name
+        self.s3_targets_configuration = s3_targets_configuration
+        self.jdbc_targets_configuration = jdbc_targets_configuration
+        self.mongo_targets_configuration = mongo_targets_configuration
+        self.dynamo_targets_configuration = dynamo_targets_configuration
+        self.glue_catalog_targets_configuration = glue_catalog_targets_configuration
+        self.cron_schedule = cron_schedule
+        self.classifiers = classifiers
+        self.table_prefix = table_prefix
+        self.update_behavior = update_behavior
+        self.delete_behavior = delete_behavior
+        self.recrawl_behavior = recrawl_behavior
+        self.lineage_settings = lineage_settings
+        self.json_configuration = json_configuration
+        self.security_configuration = security_configuration
+        self.tags = tags
+        self.overwrite = overwrite
+        kwargs['client_type'] = 'glue'
+        super().__init__(*args, **kwargs)
+
+    def list_crawlers(self) -> List:
+        """:return: Lists of Crawlers"""
+        conn = self.get_conn()
+        return conn.get_crawlers()
+
+    def get_iam_execution_role(self) -> Dict:
+        """:return: iam role for crawler execution"""
+        iam_client = self.get_client_type('iam', self.region_name)
+
+        try:
+            glue_execution_role = iam_client.get_role(RoleName=self.iam_role_name)
+            self.log.info("Iam Role Name: %s", self.iam_role_name)
+            return glue_execution_role
+        except Exception as general_error:
+            self.log.error("Failed to create aws glue crawler, error: %s", general_error)
+            raise
+
+    def initialize_crawler(self):
```

Comment:
i might consider chopping this method because initialize is a bit confusing and it doesn't really do much.

then in your operator execute you would just call
```python
crawler_name = hook.get_or_create_glue_crawler(**crawler_kwargs)
hook.start_crawler(crawler_name)
hook.await_crawler(crawler_name)
```
which i think is a tad clearer